Bowling/Bowling/FrameNoBonusRollFactory.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BowlingApp
+{
+    public class FrameNoBonusRollFactory
+    {
+        public const int totalPinsCount = 10;
+        public int PlaceCountHolder { get; set; }
+        
+        public Frame Create(int frameNumber, string bowlingMarks, int placeCount)
+        {
+            PlaceCountHolder = placeCount;
+            return ScoreRolls(frameNumber, bowlingMarks);
+        }
+
+        private Frame ScoreRolls(int frameNumber, string bowlingMarks)
+        {
+            Frame frame = new Frame();
+            frame.frameNumber = frameNumber;
+            var score = bowlingMarks.Substring(PlaceCountHolder, 1);
+
+            switch (score)
+            {
+                case (RollMarks.ZeroPinsMark):
+                    frame.firstRoll = 0;
+                    break;
+
+                case (RollMarks.StrikeMark):
+                    frame.firstRoll = totalPinsCount;
```

Comment:
A good practice is to have the guts of a `case` do very little.  In this instance, this case does 5 things.  I would encourage you to make a well-named function for each of the branches of this `switch` statement that will create and return a `Frame`.  For instance, this case can looks like:

```
case(RollMarks.StrikeMark):
   return SomethingAboutAStrikeHappening(...)
```

```
private Frame SomethingAboutAStrikeHappening(...)
{
    return new Frame {
        firstRoll: totalPinsCount,
        frameOutcome: RollTypes.Strike,
        ...
    }
}
```

You'll notice I'm returning the `Frame` object directly from the `case` statement.  There isn't a need for the `ScoreRolls` as currently written to need to know about a `Frame` object.

When you accomplish this, this function will read much more nicely:

```
private Frame ScoreRolls(int frameNumber, string bowlingMarks)
{
    WhateverTheFirstSwitchRepresents(score);

    ....
}

private Frame WhateverTheFirstSwitchRepresents(RollMarks score)
{
    switch (score):
    {
        case (RollMarks.ZeroPinsMark):
            return SomethingHappensWithZeroPins();
        case (RollMarks.StrikeMark):
            return SomethingAboutAStrikeHappening();
        case (RollMarks.SpareMark):  // <-- Notice I'm explicitly listing SpareMark instead of relying on the default case, this makes it more readable
            return SomethingAboutASpareHappening():
        default:
            break;
    }
}
```

Reply:
I removed the switch statements completely and took some guidance from you sample code to make this look a lot cleaner.